### DISCUSSION TITLE
请问心电蓝牙设备端用的什么设备

### DISCUSSION THREAD

**Discussion by masterchen** [CCXXXI/ecg_monitor#334] [Q&A] [answered]
请问心电蓝牙设备端用的什么设备？ 可以增加模拟设备吗？

**Comment by CCXXXI** [comment, marked as answer]
暂时只接了数创的HA301B，模拟设备在开发者工具里能打开

**Comment by masterchen** [comment]
@CCXXXI 谢谢及时回复， 可以介绍一下开发环境吗？ 比如 工具版本之类的，我打算试试

**Comment by CCXXXI** [comment]
最新版的 Flutter
https://flutter.cn/docs/get-started/install

**Comment by masterchen** [comment]
已经编译并运行成功， 请问一下关于模拟数据，我这边没又所说的心电采集设备，我想看看数据采集后展示出来的效果。 

 前边说“模拟设备在开发者工具里能打开”  这个能进一步介绍一下怎么操作吗？

**Reply by CCXXXI** [reply]
我的-设置-显示开发者工具 开
我的-设置-模拟设备 开
导入这个测试用数据，或者按这个 JSON 格式弄点其他数据也行：https://github.com/CCXXXI/ecg_data/blob/main/assets/data.json
设备管理 连接模拟设备

**Comment by masterchen** [comment]
APP做的很好呀，还有模拟功能，太好了！

我用你GitHub上 release的APK， 在看蓝蝶模拟器上可以安装， 按照说明打开模拟器 连接模拟器 加载数据，在实时界面里边没看到数据。
我有两个疑问麻烦解答一下：
1、在设置里边 看到 加载数据一直在转圈，是不是因为网络原因数据没加载导致实时界面看不到波形？
2、我自己编译的（debug/release）两种模式下的APK都无法在蓝叠模拟器件安装，莫非是要签名？

谢谢！

**Reply by CCXXXI** [reply]
那个加载是解析所选的 JSON 文件存入数据库，和网络无关

确实很慢，因为只是个测试用的功能所以也没去优化，我前面发的那个 8 MB 多的文件可能要加载几分钟

直接用开发者工具往数据库里写能更快一点，在调试模式下运行的时候控制台里有个调试数据库的链接

---

我不确定蓝叠模拟器的情况，不过 Android 开发一般会用 Android Studio 自带的那个 Android Emulator

**Reply by masterchen** [reply]
等了十几分钟了 😄， 是不是数据库要单独安装或者做什么处理？

“在调试模式下运行的时候控制台里有个调试数据库的链接

”  这个请详细说明一下？  我在真机加载数据也是一直转圈

**Reply by CCXXXI** [reply]
数据库应该不需要特别处理，可能发生了一些我没发现的 bug

---

在 IDEA 里运行，控制台会有这个：

![ISAR CONNECT STARTED](https://github.com/CCXXXI/ecg_monitor/assets/25550534/9048f497-0ea8-4f3d-8847-342a2e7f75bc)

**Reply by masterchen** [reply]
在Android模拟器可以了， 在蓝叠和真机上，主要是选择加载模拟采样数据 他不会弹框， 所以无法选择要加载的JSON， 但是在Android模拟可以有选择文件的弹框， 应该属于一个BUG
![1685348915010](https://github.com/CCXXXI/ecg_monitor/assets/1975340/2065c2f5-9d57-48c4-a55c-3f6756be7c35)

**Reply by CCXXXI** [reply]
选择文件的窗口都弹不出来的话，应该是 file_picker 包的问题，不是我这边能处理的了

**Reply by masterchen** [reply]
` final picked = await FilePicker.platform.pickFiles(
    type: FileType.custom,
    allowedExtensions: ["json"],
  );`

将这个函数改成 不带参数的 弹框就没问题
 final picked = await FilePicker.platform.pickFiles( );

**Reply by CCXXXI** [reply]
直接开个 Pull Request 吧

**Reply by masterchen** [reply]
> 直接开个 Pull Request 吧

说不定是我这个真机 SDK版本太低也不一定，你原来那种也是标准写法，就不pull了
